firmware/main.c

```diff
@@ -3,6 +3,9 @@
 PortData port_a;
 PortData port_b;
 
+// TODO: flash buffer could be shared with the port, because they are not used simultaneously
+USB_ALIGN u8 flash_buffer[FLASH_BUFFER_SIZE];
```

Comment:
Out of curiosity, why do you need 4 byte alignment with the flash buffer and usb pipe ring buffer?